lib/segment/benches/hnsw_search_graph.rs

```diff
@@ -1,60 +1,116 @@
 #[cfg(not(target_os = "windows"))]
 mod prof;
 
+use std::hint::black_box;
+use std::path::Path;
+
 use common::types::PointOffsetType;
 use criterion::{criterion_group, criterion_main, Criterion};
+use indicatif::{ParallelProgressIterator, ProgressStyle};
 use rand::rngs::StdRng;
-use rand::{thread_rng, SeedableRng};
+use rand::SeedableRng;
+use rayon::iter::{IntoParallelIterator as _, ParallelIterator as _};
 use segment::fixtures::index_fixtures::{random_vector, FakeFilterContext, TestRawScorerProducer};
+use segment::index::hnsw_index::graph_layers::GraphLayers;
 use segment::index::hnsw_index::graph_layers_builder::GraphLayersBuilder;
-use segment::index::hnsw_index::graph_links::GraphLinksRam;
 use segment::index::hnsw_index::point_scorer::FilteredScorer;
 use segment::spaces::simple::CosineMetric;
-use segment::vector_storage::chunked_vector_storage::VectorOffsetType;
 
-const NUM_VECTORS: usize = 100000;
+const NUM_VECTORS: usize = 1_000_000;
 const DIM: usize = 64;
 const M: usize = 16;
 const TOP: usize = 10;
 const EF_CONSTRUCT: usize = 100;
 const EF: usize = 100;
 const USE_HEURISTIC: bool = true;
 
+type Metric = CosineMetric;
+
 fn hnsw_benchmark(c: &mut Criterion) {
-    let mut rng = StdRng::seed_from_u64(42);
-    let vector_holder = TestRawScorerProducer::<CosineMetric>::new(DIM, NUM_VECTORS, &mut rng);
-    let mut group = c.benchmark_group("hnsw-index-search-group");
-    let mut rng = thread_rng();
+    let mut group = c.benchmark_group("hnsw-search-graph");
+
     let fake_filter_context = FakeFilterContext {};
 
-    let mut graph_layers_builder =
-        GraphLayersBuilder::new(NUM_VECTORS, M, M * 2, EF_CONSTRUCT, 10, USE_HEURISTIC);
-    for idx in 0..(NUM_VECTORS as PointOffsetType) {
-        let added_vector = vector_holder.vectors.get(idx as VectorOffsetType).to_vec();
-        let raw_scorer = vector_holder.get_raw_scorer(added_vector).unwrap();
-        let scorer = FilteredScorer::new(raw_scorer.as_ref(), Some(&fake_filter_context));
-        let level = graph_layers_builder.get_random_layer(&mut rng);
-        graph_layers_builder.set_levels(idx, level);
-        graph_layers_builder.link_new_point(idx, scorer);
+    // Note: make sure that vector generation is deterministic.
+    let vector_holder =
```

Comment:
Are this changes from https://github.com/qdrant/qdrant/pull/5666? Did you add something new or is it copy-paste from another PR?

Reply:
Yes, this PR is based on #5666. Only the last commit is related to this PR. Check the commit tree.